{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import requests\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 70 ICGC projects\n",
      "Files to download 60\n"
     ]
    }
   ],
   "source": [
    "projects = requests.get(\"https://dcc.icgc.org/api/v1/projects?size=1000\").json()[\"hits\"]\n",
    "print \"Found {} ICGC projects\".format(len(projects))\n",
    "\n",
    "data_types = [\"simple_somatic_mutation.open\", \"exp_array\", \"donor\"]\n",
    "\n",
    "files = [{\"name\": \"{1}.{0}.tsv.gz\".format(p[\"id\"], d),\n",
    "          \"url\": \"https://dcc.icgc.org/api/v1/download?fn=/release_22/Projects/{0}/{1}.{0}.tsv.gz\".format(p[\"id\"], d)}\n",
    "         for p in projects for d in data_types]\n",
    "files_to_download = [f for f in files if not os.path.isfile(\"/data/icgc/{}\".format(f[\"name\"]))]\n",
    "print \"Files to download\", len(files_to_download)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for f in files_to_download:\n",
    "    print(\"Downloading {}\".format(f[\"name\"]))\n",
    "    r = requests.get(f[\"url\"], allow_redirects=True, verify=False)\n",
    "    if r.status_code == requests.codes.ok:\n",
    "        with open(\"/data/icgc/{}\".format(f[\"name\"]), 'wb+') as tar:\n",
    "            tar.write(r.content)\n",
    "    else:\n",
    "        print(\"Problems downloading {}: {}\".format(f[\"name\"], r.status_code))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>donor_survival_time</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>cancer_type_prior_malignancy</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DO2</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXFI</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>2865</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2865</td>\n",
       "      <td>2920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DO1</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXDK</td>\n",
       "      <td>male</td>\n",
       "      <td>deceased</td>\n",
       "      <td>relapse</td>\n",
       "      <td>progression (liquid tumours)</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>1169</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2037</td>\n",
       "      <td>2037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DO4</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXGP</td>\n",
       "      <td>female</td>\n",
       "      <td>alive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>5087</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5087</td>\n",
       "      <td>5087</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DO3</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXFN</td>\n",
       "      <td>male</td>\n",
       "      <td>deceased</td>\n",
       "      <td>relapse</td>\n",
       "      <td>progression (liquid tumours)</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>650</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1068</td>\n",
       "      <td>1068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DO6</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXPN</td>\n",
       "      <td>male</td>\n",
       "      <td>alive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>4946</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4946</td>\n",
       "      <td>4946</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_donor_id project_code  study_donor_involved_in submitted_donor_id  \\\n",
       "0           DO2       ALL-US                      NaN   TARGET-10-PAIXFI   \n",
       "1           DO1       ALL-US                      NaN   TARGET-10-PAIXDK   \n",
       "2           DO4       ALL-US                      NaN   TARGET-10-PAIXGP   \n",
       "3           DO3       ALL-US                      NaN   TARGET-10-PAIXFN   \n",
       "4           DO6       ALL-US                      NaN   TARGET-10-PAIXPN   \n",
       "\n",
       "  donor_sex donor_vital_status disease_status_last_followup  \\\n",
       "0      male                NaN                          NaN   \n",
       "1      male           deceased                      relapse   \n",
       "2    female              alive                          NaN   \n",
       "3      male           deceased                      relapse   \n",
       "4      male              alive                          NaN   \n",
       "\n",
       "             donor_relapse_type  donor_age_at_diagnosis  \\\n",
       "0                           NaN                       5   \n",
       "1  progression (liquid tumours)                       7   \n",
       "2                           NaN                       2   \n",
       "3  progression (liquid tumours)                      15   \n",
       "4                           NaN                       1   \n",
       "\n",
       "   donor_age_at_enrollment                  ...                   \\\n",
       "0                        5                  ...                    \n",
       "1                        7                  ...                    \n",
       "2                        2                  ...                    \n",
       "3                       15                  ...                    \n",
       "4                        1                  ...                    \n",
       "\n",
       "   donor_relapse_interval  donor_diagnosis_icd10  \\\n",
       "0                    2865                  C91.0   \n",
       "1                    1169                  C91.0   \n",
       "2                    5087                  C91.0   \n",
       "3                     650                  C91.0   \n",
       "4                    4946                  C91.0   \n",
       "\n",
       "  donor_tumour_staging_system_at_diagnosis  donor_tumour_stage_at_diagnosis  \\\n",
       "0                                      NaN                              NaN   \n",
       "1                                      NaN                              NaN   \n",
       "2                                      NaN                              NaN   \n",
       "3                                      NaN                              NaN   \n",
       "4                                      NaN                              NaN   \n",
       "\n",
       "   donor_tumour_stage_at_diagnosis_supplemental  donor_survival_time  \\\n",
       "0                                           NaN                 2865   \n",
       "1                                           NaN                 2037   \n",
       "2                                           NaN                 5087   \n",
       "3                                           NaN                 1068   \n",
       "4                                           NaN                 4946   \n",
       "\n",
       "   donor_interval_of_last_followup  prior_malignancy  \\\n",
       "0                             2920               NaN   \n",
       "1                             2037               NaN   \n",
       "2                             5087               NaN   \n",
       "3                             1068               NaN   \n",
       "4                             4946               NaN   \n",
       "\n",
       "   cancer_type_prior_malignancy  cancer_history_first_degree_relative  \n",
       "0                           NaN                                   NaN  \n",
       "1                           NaN                                   NaN  \n",
       "2                           NaN                                   NaN  \n",
       "3                           NaN                                   NaN  \n",
       "4                           NaN                                   NaN  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "donors = pd.read_table(\"/data/icgc/donor.ALL-US.tsv.gz\")\n",
    "donors.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/opt/conda/envs/python2/lib/python2.7/site-packages/numpy/lib/function_base.py:3403: RuntimeWarning: Invalid value encountered in median\n",
      "  RuntimeWarning)\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>donor_age_at_last_followup</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>donor_survival_time</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>cancer_type_prior_malignancy</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>count</th>\n",
       "      <td>0.0</td>\n",
       "      <td>1002.000000</td>\n",
       "      <td>1002.000000</td>\n",
       "      <td>1002.000000</td>\n",
       "      <td>1002.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1002.000000</td>\n",
       "      <td>1002.000000</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>mean</th>\n",
       "      <td>NaN</td>\n",
       "      <td>9.927146</td>\n",
       "      <td>9.927146</td>\n",
       "      <td>16.584830</td>\n",
       "      <td>2105.264471</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2435.883234</td>\n",
       "      <td>2432.849301</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>std</th>\n",
       "      <td>NaN</td>\n",
       "      <td>5.805857</td>\n",
       "      <td>5.805857</td>\n",
       "      <td>6.196845</td>\n",
       "      <td>1288.331783</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1175.223470</td>\n",
       "      <td>1173.425799</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>min</th>\n",
       "      <td>NaN</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>1.000000</td>\n",
       "      <td>2.000000</td>\n",
       "      <td>20.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>28.000000</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25%</th>\n",
       "      <td>NaN</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>4.000000</td>\n",
       "      <td>12.000000</td>\n",
       "      <td>905.750000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1509.000000</td>\n",
       "      <td>1460.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50%</th>\n",
       "      <td>NaN</td>\n",
       "      <td>11.000000</td>\n",
       "      <td>11.000000</td>\n",
       "      <td>17.000000</td>\n",
       "      <td>2281.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2775.000000</td>\n",
       "      <td>2751.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>75%</th>\n",
       "      <td>NaN</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>15.000000</td>\n",
       "      <td>21.000000</td>\n",
       "      <td>3154.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3236.500000</td>\n",
       "      <td>3253.500000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>max</th>\n",
       "      <td>NaN</td>\n",
       "      <td>30.000000</td>\n",
       "      <td>30.000000</td>\n",
       "      <td>34.000000</td>\n",
       "      <td>5137.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5204.000000</td>\n",
       "      <td>5204.000000</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "       study_donor_involved_in  donor_age_at_diagnosis  \\\n",
       "count                      0.0             1002.000000   \n",
       "mean                       NaN                9.927146   \n",
       "std                        NaN                5.805857   \n",
       "min                        NaN                1.000000   \n",
       "25%                        NaN                4.000000   \n",
       "50%                        NaN               11.000000   \n",
       "75%                        NaN               15.000000   \n",
       "max                        NaN               30.000000   \n",
       "\n",
       "       donor_age_at_enrollment  donor_age_at_last_followup  \\\n",
       "count              1002.000000                 1002.000000   \n",
       "mean                  9.927146                   16.584830   \n",
       "std                   5.805857                    6.196845   \n",
       "min                   1.000000                    2.000000   \n",
       "25%                   4.000000                   12.000000   \n",
       "50%                  11.000000                   17.000000   \n",
       "75%                  15.000000                   21.000000   \n",
       "max                  30.000000                   34.000000   \n",
       "\n",
       "       donor_relapse_interval  donor_tumour_staging_system_at_diagnosis  \\\n",
       "count             1002.000000                                       0.0   \n",
       "mean              2105.264471                                       NaN   \n",
       "std               1288.331783                                       NaN   \n",
       "min                 20.000000                                       NaN   \n",
       "25%                905.750000                                       NaN   \n",
       "50%               2281.000000                                       NaN   \n",
       "75%               3154.000000                                       NaN   \n",
       "max               5137.000000                                       NaN   \n",
       "\n",
       "       donor_tumour_stage_at_diagnosis  \\\n",
       "count                              0.0   \n",
       "mean                               NaN   \n",
       "std                                NaN   \n",
       "min                                NaN   \n",
       "25%                                NaN   \n",
       "50%                                NaN   \n",
       "75%                                NaN   \n",
       "max                                NaN   \n",
       "\n",
       "       donor_tumour_stage_at_diagnosis_supplemental  donor_survival_time  \\\n",
       "count                                           0.0          1002.000000   \n",
       "mean                                            NaN          2435.883234   \n",
       "std                                             NaN          1175.223470   \n",
       "min                                             NaN            28.000000   \n",
       "25%                                             NaN          1509.000000   \n",
       "50%                                             NaN          2775.000000   \n",
       "75%                                             NaN          3236.500000   \n",
       "max                                             NaN          5204.000000   \n",
       "\n",
       "       donor_interval_of_last_followup  prior_malignancy  \\\n",
       "count                      1002.000000               0.0   \n",
       "mean                       2432.849301               NaN   \n",
       "std                        1173.425799               NaN   \n",
       "min                           0.000000               NaN   \n",
       "25%                        1460.000000               NaN   \n",
       "50%                        2751.000000               NaN   \n",
       "75%                        3253.500000               NaN   \n",
       "max                        5204.000000               NaN   \n",
       "\n",
       "       cancer_type_prior_malignancy  cancer_history_first_degree_relative  \n",
       "count                           0.0                                   0.0  \n",
       "mean                            NaN                                   NaN  \n",
       "std                             NaN                                   NaN  \n",
       "min                             NaN                                   NaN  \n",
       "25%                             NaN                                   NaN  \n",
       "50%                             NaN                                   NaN  \n",
       "75%                             NaN                                   NaN  \n",
       "max                             NaN                                   NaN  "
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "donors.describe()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>submitted_sample_id</th>\n",
       "      <th>analysis_id</th>\n",
       "      <th>gene_model</th>\n",
       "      <th>gene_id</th>\n",
       "      <th>normalized_expression_value</th>\n",
       "      <th>fold_change</th>\n",
       "      <th>platform</th>\n",
       "      <th>experimental_protocol</th>\n",
       "      <th>normalization_algorithm</th>\n",
       "      <th>other_analysis_algorithm</th>\n",
       "      <th>raw_data_repository</th>\n",
       "      <th>raw_data_accession</th>\n",
       "      <th>reference_sample_type</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DO2</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP2</td>\n",
       "      <td>SA4</td>\n",
       "      <td>TARGET-10-PAIXFI-03A-01R</td>\n",
       "      <td>TARGET_ALL_57_20120329</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NM_013994</td>\n",
       "      <td>1834.0310</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affymetrix Human U133 Plus 2.0</td>\n",
       "      <td>cRNA http://www.ebi.ac.uk/arrayexpress/experim...</td>\n",
       "      <td>MAS5.0</td>\n",
       "      <td>Collapsed Dataset - http://www.broadinstitute....</td>\n",
       "      <td>GEO</td>\n",
       "      <td>GSE11877</td>\n",
       "      <td>unrelated normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DO2</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP2</td>\n",
       "      <td>SA4</td>\n",
       "      <td>TARGET-10-PAIXFI-03A-01R</td>\n",
       "      <td>TARGET_ALL_57_20120329</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NM_181471</td>\n",
       "      <td>941.9643</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affymetrix Human U133 Plus 2.0</td>\n",
       "      <td>cRNA http://www.ebi.ac.uk/arrayexpress/experim...</td>\n",
       "      <td>MAS5.0</td>\n",
       "      <td>Collapsed Dataset - http://www.broadinstitute....</td>\n",
       "      <td>GEO</td>\n",
       "      <td>GSE11877</td>\n",
       "      <td>unrelated normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DO2</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP2</td>\n",
       "      <td>SA4</td>\n",
       "      <td>TARGET-10-PAIXFI-03A-01R</td>\n",
       "      <td>TARGET_ALL_57_20120329</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NM_002155</td>\n",
       "      <td>173.2684</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affymetrix Human U133 Plus 2.0</td>\n",
       "      <td>cRNA http://www.ebi.ac.uk/arrayexpress/experim...</td>\n",
       "      <td>MAS5.0</td>\n",
       "      <td>Collapsed Dataset - http://www.broadinstitute....</td>\n",
       "      <td>GEO</td>\n",
       "      <td>GSE11877</td>\n",
       "      <td>unrelated normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DO2</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP2</td>\n",
       "      <td>SA4</td>\n",
       "      <td>TARGET-10-PAIXFI-03A-01R</td>\n",
       "      <td>TARGET_ALL_57_20120329</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>XM_005263758</td>\n",
       "      <td>931.3263</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affymetrix Human U133 Plus 2.0</td>\n",
       "      <td>cRNA http://www.ebi.ac.uk/arrayexpress/experim...</td>\n",
       "      <td>MAS5.0</td>\n",
       "      <td>Collapsed Dataset - http://www.broadinstitute....</td>\n",
       "      <td>GEO</td>\n",
       "      <td>GSE11877</td>\n",
       "      <td>unrelated normal</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DO2</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP2</td>\n",
       "      <td>SA4</td>\n",
       "      <td>TARGET-10-PAIXFI-03A-01R</td>\n",
       "      <td>TARGET_ALL_57_20120329</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NM_000409</td>\n",
       "      <td>264.4506</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Affymetrix Human U133 Plus 2.0</td>\n",
       "      <td>cRNA http://www.ebi.ac.uk/arrayexpress/experim...</td>\n",
       "      <td>MAS5.0</td>\n",
       "      <td>Collapsed Dataset - http://www.broadinstitute....</td>\n",
       "      <td>GEO</td>\n",
       "      <td>GSE11877</td>\n",
       "      <td>unrelated normal</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_donor_id project_code icgc_specimen_id icgc_sample_id  \\\n",
       "0           DO2       ALL-US              SP2            SA4   \n",
       "1           DO2       ALL-US              SP2            SA4   \n",
       "2           DO2       ALL-US              SP2            SA4   \n",
       "3           DO2       ALL-US              SP2            SA4   \n",
       "4           DO2       ALL-US              SP2            SA4   \n",
       "\n",
       "        submitted_sample_id             analysis_id gene_model       gene_id  \\\n",
       "0  TARGET-10-PAIXFI-03A-01R  TARGET_ALL_57_20120329     RefSeq     NM_013994   \n",
       "1  TARGET-10-PAIXFI-03A-01R  TARGET_ALL_57_20120329     RefSeq     NM_181471   \n",
       "2  TARGET-10-PAIXFI-03A-01R  TARGET_ALL_57_20120329     RefSeq     NM_002155   \n",
       "3  TARGET-10-PAIXFI-03A-01R  TARGET_ALL_57_20120329     RefSeq  XM_005263758   \n",
       "4  TARGET-10-PAIXFI-03A-01R  TARGET_ALL_57_20120329     RefSeq     NM_000409   \n",
       "\n",
       "   normalized_expression_value  fold_change                        platform  \\\n",
       "0                    1834.0310          NaN  Affymetrix Human U133 Plus 2.0   \n",
       "1                     941.9643          NaN  Affymetrix Human U133 Plus 2.0   \n",
       "2                     173.2684          NaN  Affymetrix Human U133 Plus 2.0   \n",
       "3                     931.3263          NaN  Affymetrix Human U133 Plus 2.0   \n",
       "4                     264.4506          NaN  Affymetrix Human U133 Plus 2.0   \n",
       "\n",
       "                               experimental_protocol normalization_algorithm  \\\n",
       "0  cRNA http://www.ebi.ac.uk/arrayexpress/experim...                  MAS5.0   \n",
       "1  cRNA http://www.ebi.ac.uk/arrayexpress/experim...                  MAS5.0   \n",
       "2  cRNA http://www.ebi.ac.uk/arrayexpress/experim...                  MAS5.0   \n",
       "3  cRNA http://www.ebi.ac.uk/arrayexpress/experim...                  MAS5.0   \n",
       "4  cRNA http://www.ebi.ac.uk/arrayexpress/experim...                  MAS5.0   \n",
       "\n",
       "                            other_analysis_algorithm raw_data_repository  \\\n",
       "0  Collapsed Dataset - http://www.broadinstitute....                 GEO   \n",
       "1  Collapsed Dataset - http://www.broadinstitute....                 GEO   \n",
       "2  Collapsed Dataset - http://www.broadinstitute....                 GEO   \n",
       "3  Collapsed Dataset - http://www.broadinstitute....                 GEO   \n",
       "4  Collapsed Dataset - http://www.broadinstitute....                 GEO   \n",
       "\n",
       "  raw_data_accession reference_sample_type  \n",
       "0           GSE11877      unrelated normal  \n",
       "1           GSE11877      unrelated normal  \n",
       "2           GSE11877      unrelated normal  \n",
       "3           GSE11877      unrelated normal  \n",
       "4           GSE11877      unrelated normal  "
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "expression = pd.read_table(\"/data/icgc/exp_array.ALL-US.tsv.gz\")\n",
    "expression.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 82,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>submitted_sample_id</th>\n",
       "      <th>submitted_matched_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>...</th>\n",
       "      <th>experimental_protocol</th>\n",
       "      <th>sequencing_strategy</th>\n",
       "      <th>base_calling_algorithm</th>\n",
       "      <th>alignment_algorithm</th>\n",
       "      <th>variation_calling_algorithm</th>\n",
       "      <th>other_analysis_algorithm</th>\n",
       "      <th>seq_coverage</th>\n",
       "      <th>raw_data_repository</th>\n",
       "      <th>raw_data_accession</th>\n",
       "      <th>initial_data_release_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id icgc_sample_id  \\\n",
       "0        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "1        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "2        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "3        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "4        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "\n",
       "  matched_icgc_sample_id       submitted_sample_id  \\\n",
       "0                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "1                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "2                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "3                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "4                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "\n",
       "  submitted_matched_sample_id chromosome  chromosome_start  \\\n",
       "0    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "1    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "2    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "3    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "4    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "\n",
       "             ...              experimental_protocol  sequencing_strategy  \\\n",
       "0            ...                         Paired End                  WXS   \n",
       "1            ...                         Paired End                  WXS   \n",
       "2            ...                         Paired End                  WXS   \n",
       "3            ...                         Paired End                  WXS   \n",
       "4            ...                         Paired End                  WXS   \n",
       "\n",
       "  base_calling_algorithm alignment_algorithm variation_calling_algorithm  \\\n",
       "0                 CASAVA                 BWA                     SNVMix2   \n",
       "1                 CASAVA                 BWA                     SNVMix2   \n",
       "2                 CASAVA                 BWA                     SNVMix2   \n",
       "3                 CASAVA                 BWA                     SNVMix2   \n",
       "4                 CASAVA                 BWA                     SNVMix2   \n",
       "\n",
       "  other_analysis_algorithm seq_coverage  raw_data_repository  \\\n",
       "0                      NaN          NaN                CGHub   \n",
       "1                      NaN          NaN                CGHub   \n",
       "2                      NaN          NaN                CGHub   \n",
       "3                      NaN          NaN                CGHub   \n",
       "4                      NaN          NaN                CGHub   \n",
       "\n",
       "                     raw_data_accession  initial_data_release_date  \n",
       "0  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "1  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "2  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "3  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "4  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 82,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants = pd.read_table(\"/data/icgc/simple_somatic_mutation.open.ALL-US.tsv.gz\")\n",
    "variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'icgc_mutation_id', u'icgc_donor_id', u'project_code',\n",
       "       u'icgc_specimen_id', u'icgc_sample_id', u'matched_icgc_sample_id',\n",
       "       u'submitted_sample_id', u'submitted_matched_sample_id', u'chromosome',\n",
       "       u'chromosome_start', u'chromosome_end', u'chromosome_strand',\n",
       "       u'assembly_version', u'mutation_type', u'reference_genome_allele',\n",
       "       u'mutated_from_allele', u'mutated_to_allele', u'quality_score',\n",
       "       u'probability', u'total_read_count', u'mutant_allele_read_count',\n",
       "       u'verification_status', u'verification_platform',\n",
       "       u'biological_validation_status', u'biological_validation_platform',\n",
       "       u'consequence_type', u'aa_mutation', u'cds_mutation', u'gene_affected',\n",
       "       u'transcript_affected', u'gene_build_version', u'platform',\n",
       "       u'experimental_protocol', u'sequencing_strategy',\n",
       "       u'base_calling_algorithm', u'alignment_algorithm',\n",
       "       u'variation_calling_algorithm', u'other_analysis_algorithm',\n",
       "       u'seq_coverage', u'raw_data_repository', u'raw_data_accession',\n",
       "       u'initial_data_release_date'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 91,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "34"
      ]
     },
     "execution_count": 97,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "len(variants.groupby(\"submitted_sample_id\"))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'DataFrame' object has no attribute 'analyzed_sample_id'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m          Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-89-d861eb9e1907>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m# Submissions dict keyed by icgc_sample_id\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0msubmissions\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m{\u001b[0m\u001b[0mid\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;33m}\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mid\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mexpression\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0manalyzed_sample_id\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvariants\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0manalyzed_sample_id\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      3\u001b[0m \u001b[1;32mprint\u001b[0m \u001b[1;34m\"Found {} samples with either or both expression and variant data\"\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msubmissions\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[1;31m# for sample_id in expression.icgc_sample_id.unique():\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/opt/conda/envs/python2/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36m__getattr__\u001b[1;34m(self, name)\u001b[0m\n\u001b[0;32m   2670\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mname\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_info_axis\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2671\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2672\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2673\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2674\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'analyzed_sample_id'"
     ]
    }
   ],
   "source": [
    "# Submissions dict keyed by icgc_sample_id\n",
    "submissions = {id: {} for id in expression.analyzed_sample_id.append(variants.analyzed_sample_id).unique()}\n",
    "print \"Found {} samples with either or both expression and variant data\".format(len(submissions))\n",
    "    \n",
    "# for sample_id in expression.icgc_sample_id.unique():\n",
    "#     submissions[sample_id][\"expression\"] = expression[expression[\"icgc_sample_id\"] == sample_id][\n",
    "#         [\"gene_id\", \"normalized_expression_value\"]]\n",
    "#     submissions[sample_id][\"icgc_donor_id\"] = expression[expression[\"icgc_sample_id\"] == sample_id].iloc[0]\n",
    "\n",
    "# for sample_id in variants.icgc_sample_id.unique():\n",
    "#     submissions[sample_id][\"variants\"] = variants[variants[\"icgc_sample_id\"] == sample_id][\n",
    "#         [\"chromosome\", \"chromosome_start\", \"mutated_from_allele\", \"mutated_to_allele\"]]\n",
    "#     submissions[sample_id][\"icgc_donor_id\"] = \\\n",
    "#         variants[variants[\"icgc_sample_id\"] == sample_id][\"icgc_donor_id\"].get_value\n",
    "# submissions[\"SA565\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
