{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": [
    "import os\n",
    "import requests\n",
    "import pandas as pd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 44,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "Found 70 total ICGC projects\n",
      "Found 60 with ssm\n",
      "Files to download 0\n"
     ]
    }
   ],
   "source": [
    "projects = requests.get(\"https://dcc.icgc.org/api/v1/projects?size=1000\").json()[\"hits\"]\n",
    "print \"Found {} total ICGC projects\".format(len(projects))\n",
    "\n",
    "projects_with_ssm = [p for p in projects if \"availableDataTypes\" in p and \"ssm\" in p[\"availableDataTypes\"]]\n",
    "print \"Found {} with ssm\".format(len(projects_with_ssm))\n",
    "\n",
    "data_types = [\"donor\", \"simple_somatic_mutation.open\"]\n",
    "files = [{\"name\": \"{1}.{0}.tsv.gz\".format(p[\"id\"], d),\n",
    "          \"url\": \"https://dcc.icgc.org/api/v1/download?fn=/release_22/Projects/{0}/{1}.{0}.tsv.gz\".format(p[\"id\"], d)}\n",
    "         for p in projects_with_ssm for d in data_types]\n",
    "files_to_download = [f for f in files if not os.path.isfile(\"/data/icgc/{}\".format(f[\"name\"]))]\n",
    "print \"Files to download\", len(files_to_download)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "for f in files_to_download:\n",
    "    print(\"Downloading {}\".format(f[\"name\"]))\n",
    "    r = requests.get(f[\"url\"], allow_redirects=True, verify=False)\n",
    "    if r.status_code == requests.codes.ok:\n",
    "        with open(\"/data/icgc/{}\".format(f[\"name\"]), 'wb+') as tar:\n",
    "            tar.write(r.content)\n",
    "    else:\n",
    "        print(\"Problems downloading {}: {}\".format(f[\"name\"], r.status_code))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>donor_survival_time</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>cancer_type_prior_malignancy</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>DO2</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXFI</td>\n",
       "      <td>male</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>5</td>\n",
       "      <td>...</td>\n",
       "      <td>2865</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2865</td>\n",
       "      <td>2920</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>DO1</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXDK</td>\n",
       "      <td>male</td>\n",
       "      <td>deceased</td>\n",
       "      <td>relapse</td>\n",
       "      <td>progression (liquid tumours)</td>\n",
       "      <td>7</td>\n",
       "      <td>7</td>\n",
       "      <td>...</td>\n",
       "      <td>1169</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2037</td>\n",
       "      <td>2037</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>DO4</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXGP</td>\n",
       "      <td>female</td>\n",
       "      <td>alive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>...</td>\n",
       "      <td>5087</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5087</td>\n",
       "      <td>5087</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>DO3</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXFN</td>\n",
       "      <td>male</td>\n",
       "      <td>deceased</td>\n",
       "      <td>relapse</td>\n",
       "      <td>progression (liquid tumours)</td>\n",
       "      <td>15</td>\n",
       "      <td>15</td>\n",
       "      <td>...</td>\n",
       "      <td>650</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1068</td>\n",
       "      <td>1068</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>DO6</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PAIXPN</td>\n",
       "      <td>male</td>\n",
       "      <td>alive</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>...</td>\n",
       "      <td>4946</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4946</td>\n",
       "      <td>4946</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_donor_id project_code  study_donor_involved_in submitted_donor_id  \\\n",
       "0           DO2       ALL-US                      NaN   TARGET-10-PAIXFI   \n",
       "1           DO1       ALL-US                      NaN   TARGET-10-PAIXDK   \n",
       "2           DO4       ALL-US                      NaN   TARGET-10-PAIXGP   \n",
       "3           DO3       ALL-US                      NaN   TARGET-10-PAIXFN   \n",
       "4           DO6       ALL-US                      NaN   TARGET-10-PAIXPN   \n",
       "\n",
       "  donor_sex donor_vital_status disease_status_last_followup  \\\n",
       "0      male                NaN                          NaN   \n",
       "1      male           deceased                      relapse   \n",
       "2    female              alive                          NaN   \n",
       "3      male           deceased                      relapse   \n",
       "4      male              alive                          NaN   \n",
       "\n",
       "             donor_relapse_type  donor_age_at_diagnosis  \\\n",
       "0                           NaN                       5   \n",
       "1  progression (liquid tumours)                       7   \n",
       "2                           NaN                       2   \n",
       "3  progression (liquid tumours)                      15   \n",
       "4                           NaN                       1   \n",
       "\n",
       "   donor_age_at_enrollment                  ...                   \\\n",
       "0                        5                  ...                    \n",
       "1                        7                  ...                    \n",
       "2                        2                  ...                    \n",
       "3                       15                  ...                    \n",
       "4                        1                  ...                    \n",
       "\n",
       "   donor_relapse_interval  donor_diagnosis_icd10  \\\n",
       "0                    2865                  C91.0   \n",
       "1                    1169                  C91.0   \n",
       "2                    5087                  C91.0   \n",
       "3                     650                  C91.0   \n",
       "4                    4946                  C91.0   \n",
       "\n",
       "  donor_tumour_staging_system_at_diagnosis  donor_tumour_stage_at_diagnosis  \\\n",
       "0                                      NaN                              NaN   \n",
       "1                                      NaN                              NaN   \n",
       "2                                      NaN                              NaN   \n",
       "3                                      NaN                              NaN   \n",
       "4                                      NaN                              NaN   \n",
       "\n",
       "   donor_tumour_stage_at_diagnosis_supplemental  donor_survival_time  \\\n",
       "0                                           NaN                 2865   \n",
       "1                                           NaN                 2037   \n",
       "2                                           NaN                 5087   \n",
       "3                                           NaN                 1068   \n",
       "4                                           NaN                 4946   \n",
       "\n",
       "   donor_interval_of_last_followup  prior_malignancy  \\\n",
       "0                             2920               NaN   \n",
       "1                             2037               NaN   \n",
       "2                             5087               NaN   \n",
       "3                             1068               NaN   \n",
       "4                             4946               NaN   \n",
       "\n",
       "   cancer_type_prior_malignancy  cancer_history_first_degree_relative  \n",
       "0                           NaN                                   NaN  \n",
       "1                           NaN                                   NaN  \n",
       "2                           NaN                                   NaN  \n",
       "3                           NaN                                   NaN  \n",
       "4                           NaN                                   NaN  \n",
       "\n",
       "[5 rows x 21 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "donors = pd.read_table(\"/data/icgc/donor.ALL-US.tsv.gz\")\n",
    "donors.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'icgc_donor_id', u'project_code', u'study_donor_involved_in',\n",
       "       u'submitted_donor_id', u'donor_sex', u'donor_vital_status',\n",
       "       u'disease_status_last_followup', u'donor_relapse_type',\n",
       "       u'donor_age_at_diagnosis', u'donor_age_at_enrollment',\n",
       "       u'donor_age_at_last_followup', u'donor_relapse_interval',\n",
       "       u'donor_diagnosis_icd10', u'donor_tumour_staging_system_at_diagnosis',\n",
       "       u'donor_tumour_stage_at_diagnosis',\n",
       "       u'donor_tumour_stage_at_diagnosis_supplemental', u'donor_survival_time',\n",
       "       u'donor_interval_of_last_followup', u'prior_malignancy',\n",
       "       u'cancer_type_prior_malignancy',\n",
       "       u'cancer_history_first_degree_relative'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "donors.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>study_donor_involved_in</th>\n",
       "      <th>submitted_donor_id</th>\n",
       "      <th>donor_sex</th>\n",
       "      <th>donor_vital_status</th>\n",
       "      <th>disease_status_last_followup</th>\n",
       "      <th>donor_relapse_type</th>\n",
       "      <th>donor_age_at_diagnosis</th>\n",
       "      <th>donor_age_at_enrollment</th>\n",
       "      <th>...</th>\n",
       "      <th>donor_relapse_interval</th>\n",
       "      <th>donor_diagnosis_icd10</th>\n",
       "      <th>donor_tumour_staging_system_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis</th>\n",
       "      <th>donor_tumour_stage_at_diagnosis_supplemental</th>\n",
       "      <th>donor_survival_time</th>\n",
       "      <th>donor_interval_of_last_followup</th>\n",
       "      <th>prior_malignancy</th>\n",
       "      <th>cancer_type_prior_malignancy</th>\n",
       "      <th>cancer_history_first_degree_relative</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>491</th>\n",
       "      <td>DO162</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>NaN</td>\n",
       "      <td>TARGET-10-PALETF</td>\n",
       "      <td>female</td>\n",
       "      <td>deceased</td>\n",
       "      <td>relapse</td>\n",
       "      <td>progression (liquid tumours)</td>\n",
       "      <td>8</td>\n",
       "      <td>8</td>\n",
       "      <td>...</td>\n",
       "      <td>650</td>\n",
       "      <td>C91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>768</td>\n",
       "      <td>768</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>1 rows × 21 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "    icgc_donor_id project_code  study_donor_involved_in submitted_donor_id  \\\n",
       "491         DO162       ALL-US                      NaN   TARGET-10-PALETF   \n",
       "\n",
       "    donor_sex donor_vital_status disease_status_last_followup  \\\n",
       "491    female           deceased                      relapse   \n",
       "\n",
       "               donor_relapse_type  donor_age_at_diagnosis  \\\n",
       "491  progression (liquid tumours)                       8   \n",
       "\n",
       "     donor_age_at_enrollment                  ...                   \\\n",
       "491                        8                  ...                    \n",
       "\n",
       "     donor_relapse_interval  donor_diagnosis_icd10  \\\n",
       "491                     650                  C91.0   \n",
       "\n",
       "    donor_tumour_staging_system_at_diagnosis  donor_tumour_stage_at_diagnosis  \\\n",
       "491                                      NaN                              NaN   \n",
       "\n",
       "     donor_tumour_stage_at_diagnosis_supplemental  donor_survival_time  \\\n",
       "491                                           NaN                  768   \n",
       "\n",
       "     donor_interval_of_last_followup  prior_malignancy  \\\n",
       "491                              768               NaN   \n",
       "\n",
       "     cancer_type_prior_malignancy  cancer_history_first_degree_relative  \n",
       "491                           NaN                                   NaN  \n",
       "\n",
       "[1 rows x 21 columns]"
      ]
     },
     "execution_count": 50,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "donors[donors[\"icgc_donor_id\"] == \"DO162\"]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>icgc_mutation_id</th>\n",
       "      <th>icgc_donor_id</th>\n",
       "      <th>project_code</th>\n",
       "      <th>icgc_specimen_id</th>\n",
       "      <th>icgc_sample_id</th>\n",
       "      <th>matched_icgc_sample_id</th>\n",
       "      <th>submitted_sample_id</th>\n",
       "      <th>submitted_matched_sample_id</th>\n",
       "      <th>chromosome</th>\n",
       "      <th>chromosome_start</th>\n",
       "      <th>...</th>\n",
       "      <th>experimental_protocol</th>\n",
       "      <th>sequencing_strategy</th>\n",
       "      <th>base_calling_algorithm</th>\n",
       "      <th>alignment_algorithm</th>\n",
       "      <th>variation_calling_algorithm</th>\n",
       "      <th>other_analysis_algorithm</th>\n",
       "      <th>seq_coverage</th>\n",
       "      <th>raw_data_repository</th>\n",
       "      <th>raw_data_accession</th>\n",
       "      <th>initial_data_release_date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>MU4819276</td>\n",
       "      <td>DO5</td>\n",
       "      <td>ALL-US</td>\n",
       "      <td>SP873</td>\n",
       "      <td>SA565</td>\n",
       "      <td>SA167</td>\n",
       "      <td>TARGET-10-PAIXPH-03A-01D</td>\n",
       "      <td>TARGET-10-PAIXPH-10A-01D</td>\n",
       "      <td>12</td>\n",
       "      <td>25271550</td>\n",
       "      <td>...</td>\n",
       "      <td>Paired End</td>\n",
       "      <td>WXS</td>\n",
       "      <td>CASAVA</td>\n",
       "      <td>BWA</td>\n",
       "      <td>SNVMix2</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>CGHub</td>\n",
       "      <td>27e629e7-b3c8-4d3b-8ddc-0ff89433e96d</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 42 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "  icgc_mutation_id icgc_donor_id project_code icgc_specimen_id icgc_sample_id  \\\n",
       "0        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "1        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "2        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "3        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "4        MU4819276           DO5       ALL-US            SP873          SA565   \n",
       "\n",
       "  matched_icgc_sample_id       submitted_sample_id  \\\n",
       "0                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "1                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "2                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "3                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "4                  SA167  TARGET-10-PAIXPH-03A-01D   \n",
       "\n",
       "  submitted_matched_sample_id chromosome  chromosome_start  \\\n",
       "0    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "1    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "2    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "3    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "4    TARGET-10-PAIXPH-10A-01D         12          25271550   \n",
       "\n",
       "             ...              experimental_protocol  sequencing_strategy  \\\n",
       "0            ...                         Paired End                  WXS   \n",
       "1            ...                         Paired End                  WXS   \n",
       "2            ...                         Paired End                  WXS   \n",
       "3            ...                         Paired End                  WXS   \n",
       "4            ...                         Paired End                  WXS   \n",
       "\n",
       "  base_calling_algorithm alignment_algorithm variation_calling_algorithm  \\\n",
       "0                 CASAVA                 BWA                     SNVMix2   \n",
       "1                 CASAVA                 BWA                     SNVMix2   \n",
       "2                 CASAVA                 BWA                     SNVMix2   \n",
       "3                 CASAVA                 BWA                     SNVMix2   \n",
       "4                 CASAVA                 BWA                     SNVMix2   \n",
       "\n",
       "  other_analysis_algorithm seq_coverage  raw_data_repository  \\\n",
       "0                      NaN          NaN                CGHub   \n",
       "1                      NaN          NaN                CGHub   \n",
       "2                      NaN          NaN                CGHub   \n",
       "3                      NaN          NaN                CGHub   \n",
       "4                      NaN          NaN                CGHub   \n",
       "\n",
       "                     raw_data_accession  initial_data_release_date  \n",
       "0  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "1  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "2  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "3  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "4  27e629e7-b3c8-4d3b-8ddc-0ff89433e96d                        NaN  \n",
       "\n",
       "[5 rows x 42 columns]"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants = pd.read_table(\"/data/icgc/simple_somatic_mutation.open.ALL-US.tsv.gz\")\n",
    "variants.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Index([u'icgc_mutation_id', u'icgc_donor_id', u'project_code',\n",
       "       u'icgc_specimen_id', u'icgc_sample_id', u'matched_icgc_sample_id',\n",
       "       u'submitted_sample_id', u'submitted_matched_sample_id', u'chromosome',\n",
       "       u'chromosome_start', u'chromosome_end', u'chromosome_strand',\n",
       "       u'assembly_version', u'mutation_type', u'reference_genome_allele',\n",
       "       u'mutated_from_allele', u'mutated_to_allele', u'quality_score',\n",
       "       u'probability', u'total_read_count', u'mutant_allele_read_count',\n",
       "       u'verification_status', u'verification_platform',\n",
       "       u'biological_validation_status', u'biological_validation_platform',\n",
       "       u'consequence_type', u'aa_mutation', u'cds_mutation', u'gene_affected',\n",
       "       u'transcript_affected', u'gene_build_version', u'platform',\n",
       "       u'experimental_protocol', u'sequencing_strategy',\n",
       "       u'base_calling_algorithm', u'alignment_algorithm',\n",
       "       u'variation_calling_algorithm', u'other_analysis_algorithm',\n",
       "       u'seq_coverage', u'raw_data_repository', u'raw_data_accession',\n",
       "       u'initial_data_release_date'],\n",
       "      dtype='object')"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "variants.columns"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "collapsed": false
   },
   "outputs": [],
   "source": [
    "submissions = variants.groupby([\"icgc_donor_id\", \"icgc_sample_id\"])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 54,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "('DO162', 'SA172')\n",
      "1\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "491    female                       8\n",
      "('DO162', 'SA596')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "491    female                       8\n",
      "('DO200', 'SA577')\n",
      "4\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "576      male                       3\n",
      "('DO208', 'SA579')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "579      male                      13\n",
      "('DO210', 'SA622')\n",
      "3\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "571    female                       6\n",
      "('DO22', 'SA30')\n",
      "5\n",
      "   donor_sex  donor_age_at_diagnosis\n",
      "64      male                      14\n",
      "('DO228', 'SA629')\n",
      "3\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "570      male                      14\n",
      "('DO282', 'SA581')\n",
      "1\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "536      male                       2\n",
      "('DO376', 'SA133')\n",
      "3\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "619    female                      13\n",
      "('DO378', 'SA135')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "621      male                       8\n",
      "('DO378', 'SA394')\n",
      "4\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "621      male                       8\n",
      "('DO48', 'SA482')\n",
      "3\n",
      "   donor_sex  donor_age_at_diagnosis\n",
      "77      male                       1\n",
      "('DO49287', 'SA519680')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "203    female                       2\n",
      "('DO49292', 'SA519895')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "209    female                       7\n",
      "('DO49292', 'SA519896')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "209    female                       7\n",
      "('DO49309', 'SA519720')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "187      male                      13\n",
      "('DO49320', 'SA519735')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "176    female                      10\n",
      "('DO49320', 'SA519736')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "176    female                      10\n",
      "('DO49323', 'SA519744')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "172    female                       8\n",
      "('DO49326', 'SA519749')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "171    female                       7\n",
      "('DO49334', 'SA519763')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "160      male                       6\n",
      "('DO49341', 'SA519780')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "152      male                       2\n",
      "('DO49341', 'SA519899')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "152      male                       2\n",
      "('DO49343', 'SA519785')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "150      male                       4\n",
      "('DO49355', 'SA519812')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "139    female                      14\n",
      "('DO49357', 'SA519900')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "137    female                      15\n",
      "('DO49380', 'SA519858')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "122    female                       7\n",
      "('DO49380', 'SA519859')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "122    female                       7\n",
      "('DO49384', 'SA519868')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "119    female                       7\n",
      "('DO49384', 'SA519869')\n",
      "5\n",
      "    donor_sex  donor_age_at_diagnosis\n",
      "119    female                       7\n",
      "('DO5', 'SA565')\n",
      "5\n",
      "  donor_sex  donor_age_at_diagnosis\n",
      "5    female                       5\n",
      "('DO50', 'SA484')\n",
      "3\n",
      "   donor_sex  donor_age_at_diagnosis\n",
      "68      male                      18\n",
      "('DO74', 'SA506')\n",
      "5\n",
      "   donor_sex  donor_age_at_diagnosis\n",
      "80      male                      15\n",
      "('DO76', 'SA508')\n",
      "5\n",
      "   donor_sex  donor_age_at_diagnosis\n",
      "81    female                      12\n"
     ]
    }
   ],
   "source": [
    "for donor_sample, variants in submissions:\n",
    "    print donor_sample\n",
    "    print len(variants.head())\n",
    "    print donors[donors[\"icgc_donor_id\"] == donor_sample[0]][[\"donor_sex\", \"donor_age_at_diagnosis\"]]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 89,
   "metadata": {
    "collapsed": false
   },
   "outputs": [
    {
     "ename": "AttributeError",
     "evalue": "'DataFrame' object has no attribute 'analyzed_sample_id'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------\u001b[0m",
      "\u001b[1;31mAttributeError\u001b[0m          Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-89-d861eb9e1907>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m()\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[1;31m# Submissions dict keyed by icgc_sample_id\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m----> 2\u001b[1;33m \u001b[0msubmissions\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m{\u001b[0m\u001b[0mid\u001b[0m\u001b[1;33m:\u001b[0m \u001b[1;33m{\u001b[0m\u001b[1;33m}\u001b[0m \u001b[1;32mfor\u001b[0m \u001b[0mid\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mexpression\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0manalyzed_sample_id\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mvariants\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0manalyzed_sample_id\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0munique\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m}\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m      3\u001b[0m \u001b[1;32mprint\u001b[0m \u001b[1;34m\"Found {} samples with either or both expression and variant data\"\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mlen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0msubmissions\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      4\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m      5\u001b[0m \u001b[1;31m# for sample_id in expression.icgc_sample_id.unique():\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m/opt/conda/envs/python2/lib/python2.7/site-packages/pandas/core/generic.pyc\u001b[0m in \u001b[0;36m__getattr__\u001b[1;34m(self, name)\u001b[0m\n\u001b[0;32m   2670\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mname\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_info_axis\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2671\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m[\u001b[0m\u001b[0mname\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2672\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2673\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2674\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mAttributeError\u001b[0m: 'DataFrame' object has no attribute 'analyzed_sample_id'"
     ]
    }
   ],
   "source": [
    "# Submissions dict keyed by icgc_sample_id\n",
    "submissions = {id: {} for id in expression.analyzed_sample_id.append(variants.analyzed_sample_id).unique()}\n",
    "print \"Found {} samples with either or both expression and variant data\".format(len(submissions))\n",
    "    \n",
    "# for sample_id in expression.icgc_sample_id.unique():\n",
    "#     submissions[sample_id][\"expression\"] = expression[expression[\"icgc_sample_id\"] == sample_id][\n",
    "#         [\"gene_id\", \"normalized_expression_value\"]]\n",
    "#     submissions[sample_id][\"icgc_donor_id\"] = expression[expression[\"icgc_sample_id\"] == sample_id].iloc[0]\n",
    "\n",
    "# for sample_id in variants.icgc_sample_id.unique():\n",
    "#     submissions[sample_id][\"variants\"] = variants[variants[\"icgc_sample_id\"] == sample_id][\n",
    "#         [\"chromosome\", \"chromosome_start\", \"mutated_from_allele\", \"mutated_to_allele\"]]\n",
    "#     submissions[sample_id][\"icgc_donor_id\"] = \\\n",
    "#         variants[variants[\"icgc_sample_id\"] == sample_id][\"icgc_donor_id\"].get_value\n",
    "# submissions[\"SA565\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "collapsed": true
   },
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 2",
   "language": "python",
   "name": "python2"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 2
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython2",
   "version": "2.7.12"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 0
}
